source/lac/trilinos_sparse_matrix.cc

```diff
@@ -356,6 +356,19 @@ namespace TrilinosWrappers
 
 
 
+  SparseMatrix::SparseMatrix (SparseMatrix &&other)
+    :
+    column_space_map(std::move(other.column_space_map)),
+    matrix(std::move(other.matrix)),
+    nonlocal_matrix(std::move(other.nonlocal_matrix)),
+    nonlocal_matrix_exporter(std::move(other.nonlocal_matrix_exporter)),
+    last_action(other.last_action),
+    compressed(other.compressed)
+  {
+  }
```

Comment:
Does this leave `other` in a defined state, preferably in the state it was in after default construction?

Same for the sparsity pattern class.

Reply:
The state is different. In the default constructor, we initialize the pointers with an `Epetra_Map` and an `Epetra_FECrsMatrix` of size zero. Now the `shared_ptr ` are null. Same thing for the sparsity patttern class, the shared_ptr is null instead of pointing to object of size 0.

Reply:
I'm worried about things such as
```
  SparseMatrix m;
  ...do something with m;

  SparseMatrix mm;
  mm = std::move (m);

  m.reinit(...);
```
For the last line to work, `m` needs to be in a consistent state -- preferably the equivalent of how the object looks like when default initialized.

Reply:
The first thing we do in `reinit` (for both the sparse matrix and the sparsity pattern) is to `reset` the `shared_ptr`. So it doesn't matter if the `shared_ptr` points to an object or if it is a `nullptr`.